core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -323,8 +323,8 @@ void initialize() {
     return;
   }
 
-  char* envProfileCopy =
-      (char*)malloc(sizeof(char) * (strlen(envProfileLibrary) + 1));
+  char* envProfileCopy = static_cast<char*>(
```

Comment:
Is there any reason this isn't being stored in a `std::unique_ptr<char[]>` or even better,`std::string`?  Either of those types would alleviate the need for any casting at all.

Reply:
Fixed in [`18b8e58` (#3522)](https://github.com/kokkos/kokkos/pull/3522/commits/18b8e5872cf6f5a7d5aa0544b868bd9f7e9c52f0).